modin/engines/ray/pandas_on_ray/frame/axis_partition.py

```diff
@@ -62,13 +65,15 @@ def deploy_func_between_two_axis_partitions(
                 kwargs,
             )
             + tuple(partitions),
-            num_returns=num_splits * 3,
+            num_returns=num_splits * 4,
         )
 
     def _wrap_partitions(self, partitions):
         return [
-            self.partition_type(partitions[i], partitions[i + 1], partitions[i + 2])
-            for i in range(0, len(partitions), 3)
+            self.partition_type(
+                partitions[i], partitions[i + 1], partitions[i + 2], partitions[i + 3]
+            )
+            for i in range(0, len(partitions), 4)
         ]
```

Comment:
If I understand correctly what it does I suggest rewriting it for clarity like
`return [self.partition_type(object_id, length, width, call_queue) for (object_id, length, width, call_queue) in zip(*[iter(partitions)] * 4)]`

same goes for Dask version

Reply:
You are almost right. Correct line would be `return [self.partition_type(object_id, length, width, ip) for (object_id, length, width, ip) in zip(*[iter(partitions)] * 4)]`. It will be considered.